tar/read.c

```diff
@@ -264,8 +266,7 @@ read_archive(struct bsdtar *bsdtar, char mode, struct archive *writer)
 		}
 		if (r == ARCHIVE_FATAL)
 			break;
-		const char *p = archive_entry_pathname(entry);
-		if (p == NULL || p[0] == '\0') {
+		if (archive_entry_pathname(entry) == NULL || archive_entry_pathname(entry)[0] == '\0') {
```

Comment:
13>C:\Users\User\Development\libarchive\libarchive\tar\read.c(269): error C2143: syntax error : missing ';' before 'type'
13>C:\Users\User\Development\libarchive\libarchive\tar\read.c(270): error C2065: 'p' : undeclared identifier
13>C:\Users\User\Development\libarchive\libarchive\tar\read.c(270): warning C4047: '==' : 'int' differs in levels of indirection from 'void *'

I tried a variety of things including renaming the variable. The one thing I tried today which finally worked as I tinkered more was enclosing the function call in brackets. I wish I understood why VS was being so finnicky

		{ //Brackets so VS2012 allows compile
			const char *p = archive_entry_pathname(entry);
			if (p == NULL || p == '\0') {
				lafe_warnc(0, "Archive entry has empty or unreadable filename ... skipping.");
				bsdtar->return_value = 1;
				continue;
			}
		}